### PR TITLE
Change DNS package

### PR DESCRIPTION
**Pull request by FloofyPlasma**
# Description

This PR changes the package that the server connection uses to parse DNS records. This will improve compatibility with visionOS if there are ever attempts to bring to visionOS in the future. It also should make server pinging more reliable, especially in cases where no port is provided and it has to check SRV records.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation comments
- [x] My changes generate no new warnings

### CODE REVIEW

File: Sources/Core/Sources/Server/Ping/Pinger.swift [hunk 1]
```diff
@@ -29,46 +28,47 @@ public class Pinger: ObservableObject {
...
-        if self.shouldPing {
-          try? self.ping()
-        }
-      } catch {
-        self.isConnecting = false
-        log.trace("Failed to create server connection")
-        ThreadUtil.runInMain {
-          self.response = Result.failure(.connectionFailed(error))
+    await withTaskGroup(of: Void.self) { group in
+      group.addTask {
```
Comment (stackotter) [inline Sources/Core/Sources/Server/Ping/Pinger.swift]: I think you can ditch these two lines (and the corresponding closing braces) entirely, the wonders of structured concurrency 😎 The reason is that you call connect from a `Task` so it's already kinda in the background, the task group would just be jumping to a new task with no real effect

**Review by stackotter — CHANGES_REQUESTED** [review]
Looks great 👌 nice work figuring out how to propagate the asyncs in quite a swifty way without having used async/await before 😅

There's just one redundant thing that you can remove but other than that it all's all great.

**Comment by FloofyPlasma** [comment]
https://github.com/stackotter/delta-client/pull/197/commits/429320c39d20b3384905ccc3c81dfa4d819e125d switches the DNS resolver, this fixes the 30 second timeout issue, pretty much making DNS resolution instant and from my testing, more reliable and faster than the official java client :)

**Review by stackotter — APPROVED** [review]
Awesome, thanks for fixing this!

**Pull request merged**